### PR TITLE
chore: Update Style Dictionary transform

### LINKED ISSUE #106: Add transformer for system shadow tokens

### PR DESCRIPTION
**Pull request by RayRedGoose**
## Issue

Resolves #106 

## Summary

Changes done: 
- Colors with alpha has been updated and now shows correct values
- Parser has been updated to handle extensions transform
- `rgba` transform has been updated to replace transparent colors by `transparent` value
- Transform for `flatten-rgba` has been updated to support spaces, percentage alpha
- Filters for `font-family` and `letter-spacing` have been updated to handle new changes in types
- Transform for `font-weight` has been added to handle text variations
- Transform for `line-height` has been added to handle px values
- Shadow tokens documentation added
- Visual tests for color tokens added

## Release Category

Infrastructure

### Release Note
Bug fixing of generating wrong value for colors with alpha and replacing transparent colors by `transparent` value. New transforms has been added to handle text values of `font-weight` and px values of `line-height`. Transforms for `font-family` and `letter-spacing` have been updated to support different token types, transform for `flatten-rgba` has been updated to handle spaces and percentage alpha.

### CODE REVIEW

File: packages/canvas-tokens/utils/transformers/flatRGBAColor.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ export const flatRGBAColor = ({value}: DesignToken): string => {
     const innerRgb = b.replace(/rgba\(([^}]+)/g, (__: string, c: string) =>
       c.split(',').slice(0, 3).toString()
     );
-    return `rgba(${innerRgb},${alpha})`;
+    return alpha === '0' ? 'transparent' : `rgba(${innerRgb},${alpha})`;
```
Comment (alanbsmith) [inline packages/canvas-tokens/utils/transformers/flatRGBAColor.ts:16]: Comment: I wonder if we should check for 0 or 0% here. 🤔

File: packages/canvas-tokens-docs/stories/system/visual-testing/Color.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+};
+
+// TEXT COLORS TESTS
+export const Text = {
+  render: TextColors,
+};
+
+export const TextStatus = {
+  render: TextStatusColors,
+};
```
Comment (alanbsmith) [inline packages/canvas-tokens-docs/stories/system/visual-testing/Color.stories.tsx:100]: Thanks for adding these!

**Review by alanbsmith — APPROVED** [review]
One small comment, but this looks great. Thank you so much, @RayRedGoose!

File: packages/canvas-tokens/utils/transformers/flatRGBAColor.ts (hunk 1 above)
Comment (RayRedGoose) [reply packages/canvas-tokens/utils/transformers/flatRGBAColor.ts:16]: It's not looking for opacity with % at all. I'm suggesting do not use percentage in color, as we using regular tokens

Comment (RayRedGoose) [reply packages/canvas-tokens/utils/transformers/flatRGBAColor.ts]: I've added support for this as well

**Pull request merged**